azext_iot/tests/iothub/message_endpoint/conftest.py

```diff
@@ -0,0 +1,464 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from time import sleep
+import pytest
+from azext_iot.common.embedded_cli import EmbeddedCLI
+from azext_iot.tests.settings import DynamoSettings
+from azext_iot.tests.generators import generate_generic_id
+from typing import Optional, TypeVar
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+SubRequest = TypeVar('SubRequest')
+Mark = TypeVar('Mark')
+cli = EmbeddedCLI()
+REQUIRED_TEST_ENV_VARS = ["azext_iot_testrg"]
+settings = DynamoSettings(req_env_set=REQUIRED_TEST_ENV_VARS)
+RG = settings.env.azext_iot_testrg
+
+
+def generate_hub_id() -> str:
+    return f"test-hub-{generate_generic_id()}"[:40]
+
+
+def generate_hub_depenency_id() -> str:
+    return f"testhubdep{generate_generic_id()}"[:24]
+
+
+def get_closest_marker(request: SubRequest) -> Mark:
+    for item in request.session.items:
+        if item.get_closest_marker("hub_infrastructure"):
+            return item.get_closest_marker("hub_infrastructure")
+    return request.node.get_closest_marker("hub_infrastructure")
+
+
+def tags_to_dict(tags: str) -> dict:
+    result = {}
+    split_tags = tags.split()
+    for tag in split_tags:
+        kvp = tag.split("=")
+        result[kvp[0]] = kvp[1]
+    return result
+
+
+@pytest.fixture(scope="module")
+def provisioned_iot_hub_module(request, provisioned_user_identity_module) -> Optional[dict]:
+    result = _iot_hub_provisioner(request, provisioned_user_identity_module)
+    yield result
+    if result:
+        _iot_hub_removal(result["name"])
+
+
+@pytest.fixture(scope="module")
+def provisioned_only_iot_hub_module(request) -> Optional[dict]:
+    result = _iot_hub_provisioner(request)
+    yield result
+    if result:
+        _iot_hub_removal(result["name"])
+
+
+def _iot_hub_provisioner(request, provisioned_user_identity=None):
+    name = generate_hub_id()
+    base_create_command = f"iot hub create -n {name} -g {RG} --sku S1"
+
+    hub_marker = get_closest_marker(request)
+    desired_location = None
+    desired_tags = None
+    desired_sys_identity = False
+    desired_user_identity = False
+
+    if hub_marker:
+        desired_location = hub_marker.kwargs.get("location")
+        desired_tags = hub_marker.kwargs.get("desired_tags")
```

Comment:
Note that this gets rid of the original test tag functionality. I will need to think about how to add them back in a neat fashion